modules/UI/videolayout/LocalVideo.js

```diff
@@ -292,12 +293,14 @@ LocalVideo.prototype._onContainerClick = function(event) {
  */
 LocalVideo.prototype._updateVideoElement = function() {
     const localVideoContainer = document.getElementById('localVideoWrapper');
+    const videoTrack
+        = getLocalVideoTrack(APP.store.getState()['features/base/tracks']);
 
     ReactDOM.render(
         <Provider store = { APP.store }>
             <VideoTrack
                 id = 'localVideo_container'
-                videoTrack = {{ jitsiTrack: this.videoStream }} />
+                videoTrack = { videoTrack } />
```

Comment:
You substituted the ```this.videoStream``` reference with ```videoTrack``` from redux, because we expect them to be different in race situations. Won't it create consistency issues that ```this.videoStream``` is still used and maintained in this class? E.g.

```
if (localVideoContainer && this.videoStream.isEnded()) {
  ReactDOM.unmountComponentAtNode(localVideoContainer);
}
```

Reply:
I don't expect this.videoStream and videoTrack to be different in the race situation. Currently the setting of this.videoStream happens after redux has been updated with the correct local video track; the setting is done imperatively from calls to UI.addLocalStream -> VideoLayout -> LocalVideo and Conference.web -> VideoLayout -> LocalVideo.

For safety, and for visual code consistency, I could manually pass in videoStarted as an alternative instead of the entire redux representation of the track. What do you think? I have other more involved alternatives I could also suggest.

Ignore this part of the reply if you want; it's a rewording of the race condition in case it inspires more insight. After typing it out...it's longer than anticipated.
- LocalVideo is on the page, in the filmstrip, and redux has been told it has started playing (videoStarted is true). It is currently in a separate container from the remote videos container to allow scrolling of remote videos only.
- Local participant is displayed large video.
- User switches to tile view. Local participant is technically still on large video.
- LocalVideo is moved in the DOM so it displays as part of the remote videos container (for easier resizing).
- Play is called on the video element of LocalVideo. This is necessary to get the video to play after the move.
- User clicks on a remote video to pin.
- A large video update is queued to put the remote participant on large video.
- Tile view is exited due to a pin.
- LocalVideo is moved in the DOM back to its own container.
- Play is called on the video element of LocalVideo.
- Because VideoTrack is not aware that videoStarted is already true, redux is told again videoStarting is true.
- Middleware catches the track update and dispatches a mute update to legacy web UI. (This middleware does this for all TRACK_UPDATED actions).
- Large video receives the mute update, sees the local participant is currently on large video, dequeues the update for showing the remote participant, and keeps showing the local participant.

Reply:
I'd like to reassure that this bug is either convoluted and/or I am unable to explain it concisely.